### PR TITLE
Clean up leftover bugs, test coverage, and forgotten paths for admin user from old project

### PR DESCRIPTION
**Pull request by zachholcomb**
Update tests to raise test coverage to 100%
Add specs for all admin merchants crud, items crud, and merchant_items crud
Remove unused password_params method in password controllers
Address edge case handling for item set_defaults method
Remove unnecessary login clauses in model methods
Add merchants_items controller to handle routes for merchants items pages

### CODE REVIEW

File: app/controllers/admin/merchant_items_controller.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 class Admin::MerchantItemsController < Admin::BaseController 
   def index
-    @merchant = Merchant.find(params[:merchant_id])
+    @merchant = find_merchant
+  end
+
+  def new
+    @merchant = find_merchant
```
Comment (zachholcomb) [inline app/controllers/admin/merchant_items_controller.rb:7]: this isn't ideal but i needed a merchant for the admin user to be able to create an item for that user

File: app/controllers/admin/merchant_items_controller.rb [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+    @item = Item.new
+  end
+
+  def create
+    item = find_merchant.items.create(item_params)
+    item_check_creation(item)
```
Comment (zachholcomb) [inline app/controllers/admin/merchant_items_controller.rb:13]: this is mostly copied from merchant_items controller

File: app/controllers/admin/merchant_items_controller.rb [hunk 3]
```diff
@@ -30,7 +40,7 @@ def item_params
   end
   
   def find_item(merchant)
-    merchant.items.find(params[:id])
```
Comment (zachholcomb) [inline app/controllers/admin/merchant_items_controller.rb:33]: this item was not functioning correctly so had to update it

File: app/controllers/admin/merchants_controller.rb [hunk 4]
```diff
@@ -51,7 +51,7 @@ def update_merchant(merchant)
       flash[:success] = "Merchant successfully updated!"
     else
       flash[:error] = merchant.errors.full_messages.to_sentence
-      render :edit
+      redirect_to "/admin/merchants/#{merchant.id}/edit"
```
Comment (zachholcomb) [inline app/controllers/admin/merchants_controller.rb:54]: had to make sure all paths were directing to /admin/

File: app/controllers/admin/users_password_controller.rb [hunk 5]
```diff
@@ -16,11 +16,6 @@ def update
   end
 
   private
-
-  def password_params
```
Comment (zachholcomb) [inline app/controllers/admin/users_password_controller.rb:20]: these methods were never being called upon in the passwords controller

File: app/controllers/merchants_items_controller.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class MerchantsItemsController < ApplicationController
+   def index
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
```
Comment (zachholcomb) [inline app/controllers/merchants_items_controller.rb:5]: think this is a more restful solution than having the items controller taking every merchants items action

File: app/models/item.rb [hunk 7]
```diff
@@ -14,7 +14,7 @@ class Item <ApplicationRecord
   validates_numericality_of :inventory, greater_than_or_equal_to: 0
   
   def set_defaults
-    if self.image == ""
+    if self.image == "" || self.image.nil?
```
Comment (zachholcomb) [inline app/models/item.rb:17]: if an image is created not on the websites form, the if statement doesn't handle that, so added a check for nil clause

File: config/routes.rb [hunk 8]
```diff
@@ -91,8 +93,9 @@
       get '/items', to: 'merchant_items#index'
       get '/items/:item_id/edit', to: 'merchant_items#edit'
       patch '/items/:item_id', to: 'merchant_items#update'
-  
+      get '/items/new', to: 'merchant_items#new'
+      post '/items', to: 'merchant_items#create'
+      delete '/items/:item_id', to: 'merchant_items#destroy'
```
Comment (zachholcomb) [inline config/routes.rb:98]: admin was missing all this crud routes

File: spec/features/admin/merchant_items/create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
```
Comment (zachholcomb) [inline spec/features/admin/merchant_items/create_spec.rb:1]: All admin crud tests were either missing or improperly labeled so had to add all crud tests

**Comment by zachholcomb** [comment]
Next step is to clean up duplicate tests from tests not organized in the right folders
